pkg/contract/contract.go

```diff
@@ -289,11 +288,20 @@ func ParseSpec(
 	return name, string(abiBytes), nil
 }
 
+func idempotentSigner(
+	_ common.Address,
+	tx *types.Transaction,
+) (*types.Transaction, error) {
+	return tx, nil
+}
+
```

Comment:
why dont we name this emptySigner

Reply:
idempotent is kind of more expressive IMO. empty can refer to something that generated empty output?